File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        ArrayList<List<Integer>> l1=new ArrayList<>();
        for(int i=0;i<numRows;i++)
        {
            ArrayList<Integer> l2=new ArrayList<>();
            if(i==0)
            {
                l2.add(1);
            }
            else
            {
                l2.add(1);
                for(int j=1;j<i;j++)
                {
                    l2.add(l1.get(i-1).get(j-1)+l1.get(i-1).get(j));
                }
                l2.add(1);
            }
            l1.add(l2);
        }
        return l1;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        List=[]
        for i in range(numRows):
            li=[]
            li.append(1)
            for j in range(1,i):
                li.append(List[i-1][j-1]+List[i-1][j])
            List.append(li)
            if(i>0):
                li.append(1)
        return List
        